packages/contracts-bedrock/scripts/DeploySuperchain.s.sol

```diff
@@ -0,0 +1,246 @@
+        require(outputSet, "DeploySuperchainOutput: output not set");
+        return output;
+    }
+
+    function superchainProxyAdmin() public view returns (ProxyAdmin out) {
+        out = output.superchainProxyAdmin;
+        require(address(out) != address(0), "DeploySuperchainOutput: superchainProxyAdmin not set");
+    }
+
+    function superchainConfigImpl() public view returns (SuperchainConfig out) {
+        out = output.superchainConfigImpl;
+        require(address(out) != address(0), "DeploySuperchainOutput: superchainConfigImpl not set");
+    }
+
+    function superchainConfigProxy() public view returns (SuperchainConfig out) {
+        out = output.superchainConfigProxy;
+        require(address(out) != address(0), "DeploySuperchainOutput: superchainConfigProxy not set");
+    }
+
+    function protocolVersionsImpl() public view returns (ProtocolVersions out) {
+        out = output.protocolVersionsImpl;
+        require(address(out) != address(0), "DeploySuperchainOutput: protocolVersionsImpl not set");
+    }
+
+    function protocolVersionsProxy() public view returns (ProtocolVersions out) {
+        out = output.protocolVersionsProxy;
+        require(address(out) != address(0), "DeploySuperchainOutput: protocolVersionsProxy not set");
+    }
+}
+
+/// @notice Deploys the Superchain contracts that can be shared by many chains.
+/// We intentionally use the terms "Input" and "Output" to clearly distinguish this script from the
+/// existing ones that use terms of "Config" and "Artifacts".
+contract DeploySuperchain is Script {
+    DeploySuperchainInput dsi;
+    DeploySuperchainOutput dso;
+
+    function toAddress(address _sender, string memory _identifier) public pure returns (address) {
+        return address(uint160(uint256(keccak256(abi.encode(_sender, _identifier)))));
+    }
+
+    function init() internal {
+        // Deploy the input and output contracts.
+        // This function is a no-op on subsequent calls.
+        if (address(dsi) == address(0)) {
+            dsi = DeploySuperchainInput(toAddress(msg.sender, "optimism.DeploySuperchainInput"));
+            dso = DeploySuperchainOutput(toAddress(msg.sender, "optimism.DeploySuperchainOutput"));
+            vm.etch(address(dsi), type(DeploySuperchainInput).runtimeCode);
+            vm.etch(address(dso), type(DeploySuperchainOutput).runtimeCode);
+        }
+    }
+
+    /// @notice This entrypoint is for end-users to deploy from an input file and write to an output file.
+    function run(string memory _infile) public {
+        init();
+        dsi.loadInput(_infile);
+
+        runWithoutIO(dsi.inputs());
+
+        string memory outfile = "";
+        dso.writeOutput(outfile);
+        require(false, "DeploySuperchain: run is not implemented");
+    }
+
+    /// @notice This entrypoint is useful for e2e testing purposes, and doesn't use any file I/O.
+    function runWithoutIO(DeploySuperchainInput.Input memory _input) public {
```

Comment:
Instead of passing the memory struct, you could also pass the address of the config contract. That way it doesn't have to run through the struct packing work.